manager/controllers/app/m4dapplication_controller.go

```diff
@@ -23,36 +27,44 @@ import (
 	pc "github.com/ibm/the-mesh-for-data/pkg/policy-compiler/policy-compiler"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// OwnerLabelKey is a key to Labels map.
+	// All owned resources should be labeled using this key.
+	OwnerLabelKey string = "m4d.ibm.com.owner"
 )
 
 // M4DApplicationReconciler reconciles a M4DApplication object
 type M4DApplicationReconciler struct {
 	client.Client
-	Name           string
-	Log            logr.Logger
-	Scheme         *runtime.Scheme
-	VaultClient    *api.Client
-	PolicyCompiler pc.IPolicyCompiler
+	Name              string
+	Log               logr.Logger
+	Scheme            *runtime.Scheme
+	VaultClient       *api.Client
+	PolicyCompiler    pc.IPolicyCompiler
+	ResourceInterface ContextInterface
 }
 
 // +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=m4dapplications,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=m4dapplications/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=blueprints,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=plotters,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=core,resources=namespaces,verbs=get;list;watch;create;update;patch;delete
 
 // +kubebuilder:rbac:groups=*,resources=*,verbs=*
 
 // Reconcile reconciles M4DApplication CRD
-// It receives M4DApplication CRD and generates the Blueprint CRD
+// It receives M4DApplication CRD and selects the appropriate modules that will run
+// The outcome is either a single Blueprint running on the same cluster or a Plotter containing multiple Blueprints that may run on different clusters
 func (r *M4DApplicationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Log.WithValues("m4dapplication", req.NamespacedName)
-
 	// obtain M4DApplication resource
 	applicationContext := &app.M4DApplication{}
+	log.V(0).Info("Reconciled object: name = " + req.Name + " namespace = " + req.Namespace)
```

Comment:
Why don't you use r.Log (then you already have namespace/name values). Can't levels be set there as well?

Reply:
Actually, this was a leftover of some debug printouts, removed this one as well.